shpc/main/registry/__init__.py

```diff
@@ -171,18 +180,20 @@ def sync_from_remote(
 
         # No local registry provided, use default
```

Comment:
I think we might still miss all these checks if local is provided, but it's not a filesystem registry:
1. first check will be skipped - the path won't exist despite having a string
2. second check will be skipped - local is defined
3. we then only check if remote is a filesystem.

Reply:
Some of those checks were in `_sync`, but they're indeed more useful here. I'll move them